### PR TITLE
added schemas, testing routes, and reduced some boilerplate

### PR DESCRIPTION
**Pull request by DavidHewWing**
Updated Schemas according to:
## User Data Object
```
{
  _id: String,
  created_at: TimeStamp,
  updated_at: TimeStamp,
  profile_picture: String,
  title: String,
  badges: [enum],
  firstname: String,
  lastname: String,
  username: String,
  description: String,
  rank: [{
    contest_number: Integer,
    placing: Integer
  }]
}
```

## Contest Data Ojbect
```
{
  _id: String,
  week: String,
  contest_number: Integer,
  ranks: [String],
}
```

### Testing:
1. Get .env file (slack me or zi)
2. `yarn dev`
3. Go to http://localhost:8080/api/users/status and http://localhost:8080/api/contests/status

### CODE REVIEW

File: src/controllers/contest.controller.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const Contest = require('../models/contest.model');
+
+exports.status = async (req, res, next) => {
+  try {
+    return res.json({ message: 'Contest endpoint is working.'})
```
Comment (ZGao28) [inline src/controllers/contest.controller.js]: use callback error handlers for JS instead of try catch

File: src/controllers/user.controller.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const User = require('../models/user.model');
+
+exports.status = async (req, res, next) => {
+  try {
```
Comment (ZGao28) [inline src/controllers/user.controller.js]: use callback error handler

File: src/controllers/contest.controller.js (hunk 1 above)
Comment (callmekungfu) [reply src/controllers/contest.controller.js]: Try catch is usually fine, express can also automatically handle errors thrown, so there is no need for explicit calling of `next`.

When we are dealing with async functions however we might need a defined rule when it comes to handling promise resolutions (`await` or `promise.then()`).

Depending on these different ways we can either use try catch or `promise.catch()`.

https://expressjs.com/en/guide/error-handling.html

Comment (DavidHewWing) [reply src/controllers/contest.controller.js]: Going to go with Yong here, he has express expertise. Updating the linting and CI process in another PR.

**Pull request merged**